block.go

```diff
@@ -279,6 +277,12 @@ func (b *ProtocolBlock) ForEachParent(consumer func(parent Parent)) {
 }
 
 func (b *ProtocolBlock) WorkScore(workScoreStructure *WorkScoreStructure) (WorkScore, error) {
+	workScoreBytes, err := workScoreStructure.DataKilobyte.Multiply(b.Size())
```

Comment:
WorkScore seems to be calculated twice - once here because of `b.Size()` which includes Header, Signature and the internal Block and again below when we retrieve WorkScore separately. That seems incorrect.

Reply:
We only apply the data factor once for the highest level object, the protocol block, because this allows us to avoid losing precision when we divide by 1024 for the kilobyte factor.